pkg/agent/conversation.go

```diff
@@ -230,9 +230,64 @@ func (a *Conversation) RunOneRound(ctx context.Context, query string) error {
 				return fmt.Errorf("building tool call: %w", err)
 			}
 
+			// Check if this is a kubectl or bash command that might be interactive
+			isInteractive := false
+			var errMsg string
+			if call.Name == "kubectl" || call.Name == "bash" {
+				if command, ok := call.Arguments["command"].(string); ok {
+					isInteractive, errMsg = tools.IsInteractiveCommand(command)
+				}
+			}
+
+			// If interactive, show error and skip to next iteration
+			if isInteractive {
+				// Use ErrorBlock for interactive command errors
+				errorBlock := ui.NewErrorBlock().SetText(fmt.Sprintf("  %s\n", errMsg))
+				a.doc.AddBlock(errorBlock)
+
+				currChatContent = append(currChatContent, gollm.FunctionCallResult{
```

Comment:
So models that doesn't support tool-use yet, we need to condition this logic on `EnableToolUseShim` (deleted lines in line 274....).

So if `EnableToolUseShim` --> Just append as text response else proper FunctionCallResult (like you have today).


I think `EnableToolUseShim` can be pushed at the gollm layer, but that is some for other day. But I am also hoping that we will probably drop this soon because most models will support tool-use by default.